bpf/bpf_host.c

```diff
@@ -1298,12 +1298,46 @@ int cil_from_host(struct __ctx_buff *ctx)
 	return handle_netdev(ctx, true);
 }
 
+#if defined(ENABLE_IPSEC) && defined(ENABLE_IPV4) && defined(TUNNEL_MODE)
+/*
+ * If the traffic is indeed overlay traffic and it should be encrypted
+ * CTX_ACT_REDIRECT is returned, unless an error occurred, and the caller can
+ * return this code to TC.
+ *
+ * CTX_ACT_OK is returned if the traffic should continue normal processing.
+ *
+ * IS_ERR can be used to determine if this function ran into an error.
+ */
+static __always_inline int do_encrypt_overlay(struct __ctx_buff *ctx,
```

Comment:
Ah, I never liked the 'maybe' wordage since the return value should really tell you 'what' happened. But if this is preferred or a common semantic I can change it.

Reply:
:+1: I guess this is maybe more preference than anything then.

When I compared with `do_decrypt()` which steals the packet away, I found it slightly surprising that actually only some specific vxlan traffic with `ENCRYPTED_OVERLAY_ID` will be handed by this function; `do_encrypt_overlay()` seems pretty decisive/final. In the back of my mind I can also envision the next developer to come along and inspect this code to think that this is the primary ipsec encryption path.. but that said, I don't think a function rename will help with that.

I'll resolve this, doesn't seem particularly notable to resolve one way or another.